tfjs-backend-webgpu/src/shader_preprocessor.ts

```diff
@@ -142,18 +164,44 @@ export function makeShader(
     ].join('\n');
   }
 
+  let preMemberIsStruct = false;
+  let currentMemberIsStruct = false;
   let uniformDeclaration = 'struct Uniforms { NAN : f32, ';
   program.variableNames.forEach((x, i) => {
-    uniformDeclaration += `${x.charAt(0).toLowerCase() + x.slice(1)}Shape : ${
-        getCoordsDataType(inputInfo[i].shape.length)}, `;
+    const perDataType = getCoordsDataType(inputInfo[i].shape.length);
+    if (perDataType === 'vec5' || perDataType === 'vec6') {
+      currentMemberIsStruct = true;
+    }
+    if (preMemberIsStruct || currentMemberIsStruct) {
+      uniformDeclaration += `@align(16) `;
+    }
+    preMemberIsStruct = currentMemberIsStruct;
+    uniformDeclaration +=
+        `${x.charAt(0).toLowerCase() + x.slice(1)}Shape : ${perDataType}, `;
   });
-  uniformDeclaration +=
-      `outShape : ${getCoordsDataType(outputData.shape.length)}, `;
+  const outputDataType = getCoordsDataType(outputData.shape.length);
+  currentMemberIsStruct =
+      outputDataType === 'vec5' || outputDataType === 'vec6';
+  if (preMemberIsStruct || currentMemberIsStruct) {
+    uniformDeclaration += `@align(16) `;
+  }
+  preMemberIsStruct = currentMemberIsStruct;
+  uniformDeclaration += `outShape : ${outputDataType}, `;
   const stridesLength = outputData.shape.length - 1;
+  const stridesDataType = getCoordsDataType(stridesLength);
+  currentMemberIsStruct =
+      stridesDataType === 'vec5' || stridesDataType === 'vec6';
+  if (preMemberIsStruct || currentMemberIsStruct) {
+    uniformDeclaration += `@align(16) `;
+  }
+  preMemberIsStruct = currentMemberIsStruct;
   uniformDeclaration += `
-       outShapeStrides: ${getCoordsDataType(stridesLength)}, `;
+       outShapeStrides: ${stridesDataType}, `;
 
   if (program.size) {
```

Comment:
What if `preMemberIsStruct` and `program.uniforms` are true, but `program.size` is false?